### ISSUE TITLE
Stuck at Disable telemetry

### ISSUE THREAD

**Issue by sanurai6x10** [an-anime-team/the-honkers-railway-launcher#57]
I tried downloading the game, I downloaded everything, but I'm not stuck at the screen saying disable telemetry, I tried clicking on it and entering my password, but nothing happens

**Comment by danielramke** [comment]
I have the same problem at this point. (Nobara Linux 38 with GNOME).

**Comment by krypt0nn** [comment]
Must be fixed in latest versions

**Comment by danielramke** [comment]
Yes the Telemetry is now not more stuck but i see only an black screen after starting. Sound is playing but nothing is displaying. Do you have any idea what the reason could be? Graphics card is an AMD RX 6950XT Wine version: Wine-GE-Proton 8-11

**Comment by Koko2110** [comment]
I had this just now too, but managed to get around it by enabling my polkit agent. It prompted me for my sudo pass and it was all good.

If you're like me on a window manager (i3wm) without any sort of authentication agent running without explicitly starting it, this may help.

**Comment by Outlift** [comment]
hey I'm having the same problem. I'm stuck in the disable telemetry. (Arch Linux, KDE plasma)

**Comment by exepta** [comment]
@Koko2110 Thanks but it is still black :D. Here my specs for more information:
Window manager: Wayland (GNOME 44.2)
Operating System: Nobara Linux 38
Kernel: Linux 6.3.12-205.fsync.fc38.x86_64
Graphics: AMD Radeon™ RX 6950 XT (Driver version: 23.1.4)
CPU: AMD Ryzen™ 9 5900X × 24
RAM: 32GB 4000Mhz CL18

**Comment by tn5421** [comment]
I installed Honkers Railway Launcher from Lutris, and have the same issue.

Lutris log says:

```
0.0.0.0 log-upload-os.hoyoverse.com
0.0.0.0 sg-public-data-api.hoyoverse.com
 >> /etc/hosts
  [2m2023-08-16T03:15:16.367373Z[0m [33m WARN[0m [1;33manime_game_core::games::star_rail::telemetry[0m[33m: [33mServer is not disabled: log-upload-os.hoyoverse.com[0m
    [2;3mat[0m /var/home/observer/.cargo/git/checkouts/anime-game-core-31370ac61699cd7f/e8b9956/src/games/star_rail/telemetry.rs:21
```

Debugging information:

```
Anime Launcher SDK: 1.8.2
Anime Game Core: 1.13.1

GTK: 4.10.5
libadwaita: 1.3.4
pango: 1.50.14
cairo: 1.17.8
```

Opensuse Tumbleweed, Kernel 6.4.9.1, KDE Plasma 5.27.7, Wayland, AMD processor and graphics card

**Comment by dmitsuki** [comment]
> @Koko2110 Thanks but it is still black :D. Here my specs for more information: Window manager: Wayland (GNOME 44.2) Operating System: Nobara Linux 38 Kernel: Linux 6.3.12-205.fsync.fc38.x86_64 Graphics: AMD Radeon™ RX 6950 XT (Driver version: 23.1.4) CPU: AMD Ryzen™ 9 5900X × 24 RAM: 32GB 4000Mhz CL18
Try changing DXVK to the gplaysync version. That removed the black screen for me.

**Comment by Raikuyami** [comment]
Has anyone found a fix for this part ? I am stuck at this "disable telemetry" (I am pretty much new to linux, so I don't know much)

**Comment by rezzubs** [comment]
Similar to a previous comment, I had to edit /etc/hosts and add the following

```
0.0.0.0 hkrpg-log-upload-os.hoyoverse.com
0.0.0.0 log-upload-os.hoyoverse.com
```

These URLs were taken from warning messages printed by the program. For anyone doing this in the future, these might change, check the warnings.

**Comment by toasties-with-toast** [comment]
I have the same problem, has anyone found a fix yet?

**Comment by PrometheusFolly** [comment]
I am stuck on the disable telemetry screen and I am unsure where to go from here does anyone know a way past this?

**Comment by CyntexMore** [comment]
Fixed by adding the following to my `configuration.nix`:

```nix
  networking = {
    hostName = "nixos";
    networkmanager.enable = true;

    extraHosts = ''
      0.0.0.0 hkrpg-log-upload-os.hoyoverse.com
      0.0.0.0 log-upload-os.hoyoverse.com
      0.0.0.0 sg-public-data-api.hoyoverse.com
    '';
  };
```

**Comment by L1xa1s** [comment]
I was able to fix it. You have to run the launcher through the console, and when you click the "disable telemetry" button, you will have to enter the password for your PC in the console because the program accesses files that require root rights.
